src/com/google/cose/OkpSigningKey.java

```diff
@@ -178,4 +181,42 @@ public static OkpSigningKey parse(byte[] keyBytes) throws CborException, CoseExc
   public static OkpSigningKey decode(DataItem cborKey) throws CborException, CoseException {
     return new OkpSigningKey(cborKey);
   }
+
+  public byte[] sign(Algorithm algorithm, byte[] message) throws CoseException {
+    if (privateKeyBytes == null) {
+      throw new CoseException("Missing key material for signing.");
+    }
+    if (algorithm != Algorithm.SIGNING_ALGORITHM_EdDSA) {
+      throw new CoseException("Incompatible key type.");
+    }
+    return tinkSign(message);
+  }
+
+  private byte[] tinkSign(byte[] message) throws CoseException {
+    try {
+      Ed25519Sign signer = new Ed25519Sign(privateKeyBytes);
+      return signer.sign(message);
+    } catch (GeneralSecurityException e) {
+      throw new CoseException("Error while signing message.", e);
+    }
+  }
+
+  public void verify(Algorithm algorithm, byte[] message, byte[] signature) throws CoseException {
+    if (publicKeyBytes == null) {
```

Comment:
We should _always_ have the public key, right?

Even if it was not supplied, it can be derived from the private key.

Reply:
added functions for deriving the key if it is not present.